server/zoom/oauth.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package zoom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const OAuthPrompt = "[Click here to link your Zoom account.](%s/plugins/zoom/oauth2/connect?channelID=%s)"
+
+type OAuthInfo struct {
+	ZoomEmail  string
+	OAuthToken *oauth2.Token // Zoom OAuth Token, ttl 15 years
+	UserID     string        // Mattermost userID
+	ZoomID     string        // Zoom userID
+}
+
+type OAuthClient struct {
+	info      *OAuthInfo
+	config    *oauth2.Config
+	siteURL   string
+	channelID string
+	apiURL    string
+}
+
+func NewOAuthClient(info *OAuthInfo, config *oauth2.Config, siteURL, channelID, apiURL string) *OAuthClient {
+	return &OAuthClient{info, config, siteURL, channelID, apiURL}
+}
+
+func (c *OAuthClient) GetUser(userID string) (*User, *AuthError) {
+	user, err := GetUserViaOAuth(c.info.OAuthToken, c.config, c.apiURL)
+	if err != nil {
+		return nil, &AuthError{fmt.Sprintf(OAuthPrompt, c.siteURL, c.channelID), err}
+	}
+
+	return user, nil
+}
+
+func (c *OAuthClient) GetMeeting(meetingID int) (*Meeting, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
Do you mind extracting time.Second*10 as a constant? That way is easier to track and modify in case we want to do it.